src/fundus/scraping/crawler.py

```diff
@@ -106,10 +114,220 @@ def pool_queue_iter(handle: MapResult[Any], queue: Queue[_T]) -> Iterator[_T]:
+                will be yielded.
+            delay (Optional[Union[float, Delay]]): Set a delay time in seconds to be used between article
+                downloads. You can set a delay directly using float or any callable satisfying the Delay
+                protocol. If set to None, no delay will be used between batches. See Delay for more
+                information. Defaults to None.
+            threading (bool): If True, the crawler will use a dedicated thread per publisher, if set to False,
+                the crawler will use a single thread for a publishers and load articles succesively. This will greatly
+                influence performance, and it is highly recommended to use a threaded crawler. Deafults to True.
+        """
+
+        if not publishers:
+            raise ValueError("param <publishers> of <Crawler.__init__> has to be non empty")
+
+        super().__init__(*publishers)
+
+        self.restrict_sources_to = restrict_sources_to
+        self.delay = delay
+        self.threading = threading
+
+    def _fetch_articles(
+        self,
+        publisher: PublisherEnum,
+        error_handling: Literal["suppress", "catch", "raise"],
+        extraction_filter: Optional[ExtractionFilter] = None,
+        url_filter: Optional[URLFilter] = None,
+    ) -> Iterator[Article]:
+        def build_delay() -> Optional[Delay]:
+            if isinstance(self.delay, float):
+                delay = self.delay
+
+                def constant_delay() -> float:
+                    return delay
+
+                return constant_delay
+
+            elif isinstance(self.delay, Delay):
+                return self.delay
+
+            else:
+                raise TypeError("param <delay> of <Crawler.__init__>")
+
+        scraper = WebScraper(publisher, self.restrict_sources_to, build_delay())
+        yield from scraper.scrape(error_handling, extraction_filter, url_filter)
+
+    @staticmethod
+    def _single_crawl(
+        publishers: Tuple[PublisherEnum, ...], article_task: Callable[[PublisherEnum], Iterator[Article]]
+    ) -> Iterator[Article]:
+        article_iterators = [article_task(publisher) for publisher in publishers]
+        while article_iterators:
+            tmp = article_iterators
+            for iterator in tmp:
+                try:
+                    yield next(iterator)
+                except StopIteration:
+                    article_iterators.remove(iterator)
+
+    @staticmethod
+    def _threaded_crawl(
+        publishers: Tuple[PublisherEnum, ...], article_task: Callable[[PublisherEnum], Iterator[Article]]
+    ) -> Iterator[Article]:
+        article_queue: Queue[Article] = Queue(len(publishers))
+        wrapped_article_task = queue_wrapper(article_queue, article_task)
+
+        with ThreadPool(processes=len(publishers) or None) as pool:
+            yield from pool_queue_iter(pool.map_async(wrapped_article_task, publishers), article_queue)
```

Comment:
Just a minor question: We do not need the dill wrapper for the standard Crawler since it is thread based. Is that correct?

Reply:
Yes :)